### PR TITLE
notify: redact more secret data from logs

### PR DESCRIPTION
**Pull request by simonpasquier**
Follow-up of https://github.com/prometheus/alertmanager/pull/1822

### CODE REVIEW

File: notify/impl.go [hunk 1]
```diff
@@ -1210,8 +1210,7 @@ func (n *Pushover) Notify(ctx context.Context, as ...*types.Alert) (bool, error)
 		return false, err
 	}
 
-	apiURL := "https://api.pushover.net/1/messages.json"
-	u, err := url.Parse(apiURL)
+	u, err := url.Parse(n.apiURL)
```
Comment (brian-brazil) [inline notify/impl.go]: We've using this on line 1218 too

Comment (simonpasquier) [reply notify/impl.go]: Thanks!

File: notify/impl_test.go [hunk 2]
```diff
@@ -74,6 +74,73 @@ func TestPagerDutyRetryV2(t *testing.T) {
...
+		&config.PagerdutyConfig{
+			ServiceKey: config.Secret(key),
+			HTTPConfig: &commoncfg.HTTPClientConfig{},
+		},
+		createTmpl(t),
+		log.NewNopLogger(),
+	)
+	notifier.apiV1 = srv.URL
+
+	ok, err := notifier.Notify(ctx, []*types.Alert{}...)
```
Comment (mxinden) [inline notify/impl_test.go]: This seems to be the same for each added test. Could this be a table-driven test with [subtests](https://blog.golang.org/subtests) instead? 

Let me know if you want to delay it, as it is not necessary for the security patch release.

Comment (simonpasquier) [reply notify/impl_test.go]: Rather than table-driven tests (to keep each test focused on one receiver), I'll create a helper function.

Comment (mxinden) [reply notify/impl_test.go]: Sounds good, thanks!

Comment (simonpasquier) [reply notify/impl_test.go]: Done!

**Pull request merged**